packages/headless/src/controllers/core/pager/headless-core-pager.test.ts

```diff
@@ -91,70 +91,70 @@ describe('Pager', () => {
     initialState.page = 2;
     initPager();
 
-    expect(engine.actions).toContainEqual(registerPage(2));
+    expect(engine.state.pagination?.firstResult).toEqual(10);
```

Comment:
Mmm... I'm not sure I agree with this approach.

In general, what I believe we should do is:

1. Test that controller methods affecting the Redux state are dispatching the correct actions with the correct payloads
3. Test that the reducers are handling dispatched actions as expected (i.e., that the state is modified correctly).

It seems to me like what you're doing here is testing the reducer behavior directly in the controller tests. I don't think this is right.

Reply:
As discussed, I adapted my draft to represent the conclusion of our rubberducking session